### PR TITLE
[full-ci] Refactor CreateAndUpload into composables

### PR DESCRIPTION
**Pull request by lookacat**
## Description
Refactor CreateAndUpload actions into composables

## Related Issue

- https://github.com/owncloud/web/pull/8921

## Motivation and Context
In order to use "create new file" (type: xyz) or "create new folder" on multiple places we refactored them into composables
As needed in the linked whitespace contextmenu

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Technical debt
- [ ] Tests

## Checklist:

- [x] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [ ] Fix / rework unittests tests

### CODE REVIEW

File: packages/web-app-files/src/components/FilesList/ContextActions.vue [hunk 1]
```diff
@@ -49,6 +51,13 @@ export default defineComponent({
 
     const { editorActions, loadExternalAppActions } = useFileActions()
 
+    const { actions: createNewFile } = useFileActionsCreateNewFile({
```
Comment (lookacat) [inline packages/web-app-files/src/components/FilesList/ContextActions.vue:54]: this is just for demonstration purposes only to test if the action works

**Review by JammingBen — COMMENTED** [review]
I love where this is going!

File: packages/web-app-files/src/composables/actions/files/useFileActionsCreateNewFile.ts [hunk 2]
```diff
@@ -239,8 +237,24 @@ export const useFileActionsCreateNewFile = ({
         ext: newFileHandler.ext
       })
     }
-
-    // TODO: Same for mimetypesAllowedForCreation
+    console.log('mimetypesAllowedForCreation', mimetypesAllowedForCreation)
```
Comment (JanAckermann) [inline packages/web-app-files/src/composables/actions/files/useFileActionsCreateNewFile.ts:240]: dev leftover?

**Review by JanAckermann — CHANGES_REQUESTED** [review]

File: packages/web-app-files/src/composables/actions/files/useFileActionsCreateNewFile.ts [hunk 3]
```diff
@@ -0,0 +1,265 @@
...
+      onInput: checkInputValue
+    }
+
+    store.dispatch('createModal', modal)
+  }
+
+  const actions = computed((): FileAction[] => {
+    const actions = []
+    for (const newFileHandler of newFileHandlers || []) {
+      const addAppProviderFile = false
```
Comment (JammingBen) [inline packages/web-app-files/src/composables/actions/files/useFileActionsCreateNewFile.ts:224]: I believe you can even remove `addAppProviderFile` and just pass `openAction`. In `handler` where we check `addAppProviderFile ? ...` we can simply check `!openAction ? ...`

**Review by JammingBen — CHANGES_REQUESTED** [review]
Few smaller things from my side, looks quite good overall.

Types need to be improved, but that can be done in a follow-up IMO because they should be added and used globally (talking about `newFileHandlers` and `mimetypesAllowedForCreation`).

Also, we definitely need to test this with a WOPI instance before merging.

File: packages/web-app-files/src/components/AppBar/CreateAndUpload.vue [hunk 4]
```diff
@@ -25,37 +25,38 @@
...
-                appearance="raw"
-                @click="showCreateResourceModal(false, mimetype.ext, false, true)"
-              >
-                <oc-resource-icon :resource="getIconResource(mimetype)" size="medium" />
-                <span v-text="$gettextInterpolate($gettext('%{name}'), { name: mimetype.name })" />
+              <oc-button appearance="raw" @click="mimeTypeAction.handler">
+                <oc-resource-icon :resource="getIconResource(mimeTypeAction)" size="medium" />
+                <span
+                  v-text="
+                    $gettextInterpolate($gettext('%{name}'), { name: mimeTypeAction.label() })
```
[the remark this answers is not in the document]
Comment (lookacat) [reply packages/web-app-files/src/components/AppBar/CreateAndUpload.vue:57]: yes right

**Comment by ownclouders** [comment]
Results for <strong>acceptance oCIS</strong> https://drone.owncloud.com/owncloud/web/35355/58/1
<details><summary>:boom: The acceptance tests failed on retry. Please find the screenshots inside ...</summary>

<p>


### webUITextEditor-textFile_feature-L13.png
  


</p></details>

File: packages/web-app-files/src/components/SideBar/Shares/Links/CreateQuickLink.vue
Comment (JammingBen) [inline packages/web-app-files/src/components/SideBar/Shares/Links/CreateQuickLink.vue:1]: This file looks like it slipped in accidentally.

**Review by JammingBen — COMMENTED** [review]
One last nitpick from my side, rest LGTM!

**Pull request merged**